src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -212,8 +218,31 @@ ss::future<> service::configure() {
           principal,
           security::acl_host::wildcard_host(),
           security::acl_operation::all,
-          security::acl_permission::allow}}},
-      5s);
+          security::acl_permission::allow}}};
+
+    auto create_acls_stop_time = model::time_from_now(30s);
```

Comment:
issue: What happens if this fails? For example if the controller takes a minute to start.

Reply:
If the controller takes longer than 30s to start, then this will break SR; however, the alternative is to either:
1. Remove the timeout and allow the loop to continue indefinitely (this is not ideal)
2. Figure out another stop condition
3. Background the retry loop and let the loop continue indefinitely

Point 1 is not ideal since it blocks startup. I've already spoken [here](https://github.com/redpanda-data/redpanda/pull/11563#discussion_r1248095266) as to why Point 3 is not doable at this time for a bug fix. Point 2 is preferrable but I do not see what another stop condition could be beyond an arbitrary timeout. I am open to ideas.

Reply:
Adding to my reasoning for avoiding Point 3 at this time.

When I put SR startup or the retry loop in the background, many of our Ducktape tests failed due to timeouts where the requests library expected to recieve a response from the server. Of course we could re-write tests to retry those REST requests; however, that will take more time and is a large code churn for a bug fix that is blocking release.

Reply:
What if this was all driven by the oneshot so so that it may fail on startup, but the next request will either wait or try the whole lot again? It means exposing that somehow so that schema ID validation can hook it, rather than calling `_api->_sequencer.local().read_sync();`

Reply:
Probably not a great idea, but what about handlign this in the `service::mitigate_error` call: https://github.com/redpanda-data/redpanda/blob/b6e63b8af38ade6c059439d8fb0eee619b5410f6/src/v/pandaproxy/schema_registry/service.cc#L219-L238

Could fire off the one shot if we get an error?

Reply:
That is a good idea to try out but it does not sound like a simple change since we need to verify that schema ID validation will still work.

Reply:
> It means exposing that somehow so that schema ID validation can hook it

If we were to make this change, then where would we call the method for schema ID validation?